### ISSUE TITLE
TLS Error Mikrotik

### ISSUE THREAD

**Issue by infiniti8** [xtrime-ru/antizapret-vpn-docker#99]
HI!

How I can disable the TLS? I have some errors on my mikrotik, like TLS Error: unknown opcode received (30) because Mikrotik does not support TLS

**Comment by infiniti8** [comment]
редактирование флага OPENVPN_TLS_CRYPT=0 в dockeк-compose.override не помогает

логи с контейнера:

```
antizapret-vpn  | Oct 09 12:58:35 openvpn[165]: 194.226.32.124:35368 TLS Error: Unroutable control packet received from [AF_INET]194.226.32.124:35368 (si=3 op=P_CONTROL_V1)
antizapret-vpn  | Oct 09 12:58:35 openvpn[165]: read UDPv4 [ECONNREFUSED|ECONNREFUSED|ECONNREFUSED|ECONNREFUSED|ECONNREFUSED|ECONNREFUSED]: Connection refused (fd=5,code=111)
```

**Comment by xtrime-ru** [comment]
Ошибки про opcode из за обфускации трафика. Надо ждать, пока не произодет коннект: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/Dockerfile#L77

**Comment by xtrime-ru** [comment]
Есть ветка с выбором более легкой обфускации: https://github.com/xtrime-ru/antizapret-vpn-docker/pull/86
Но там надо пересобирать исходники с разными переменными. Я пока не успел сделать выбор на основе env на лету.

**Comment by travuga77** [comment]
Каково примерное время ожидания подключения?

**Comment by infiniti8** [comment]
> Каково примерное время ожидания подключения?

У меня микротик вообще не подключается, хоть сутки жди. Пришлось перейти на схему с вайргвардом

**Comment by gorbunovav** [comment]
Тут скрипт обновился:
https://github.com/GubernievS/AntiZapret-VPN/blob/main/setup/root/patch-openvpn.sh

Появился второй вариант алгоритма - "Error-free".
С ним Mikrotik подключается

**Comment by gorbunovav** [comment]
(столкнулся еще с этой проблемой:
https://forum.mikrotik.com/viewtopic.php?t=197500
соединение не устанавливается, если не подождать 30 секунд после отключения
)

**Comment by alexunderboots** [comment]
> Тут скрипт обновился: https://github.com/GubernievS/AntiZapret-VPN/blob/main/setup/root/patch-openvpn.sh
> 
> Появился второй вариант алгоритма - "Error-free". С ним Mikrotik подключается

Подскажите пожалуйста, достаточно-ли выполнить этот патч в контейнере или нет? И правильно-ли я понимаю что он для UDP?

**Comment by gorbunovav** [comment]
@alexunderboots проще видимо применить изменения в Dockerfile из PR автора (#86) и пересобрать образ локально - так изменения сохранятся в случае пере-создания контейнера

**Comment by alexunderboots** [comment]
> @alexunderboots проще видимо применить изменения в Dockerfile из PR автора (#86) и пересобрать образ локально - так изменения сохранятся в случае пере-создания контейнера

Запустил в контейнере, перевел конфиг на udp
ошибки из микрота
AZ-wg-ab: disconnected <TLS error: handshake timed out (6)>

ошибки из контейнера
TLS Error: Unroutable control packet received from [AF_INET]ххх.ххх.ххх.ххх:34381 (si=3 op=P_CONTROL_V1)

**Comment by gorbunovav** [comment]
@alexunderboots https://github.com/xtrime-ru/antizapret-vpn-docker/issues/99#issuecomment-2424164290

**Comment by alexunderboots** [comment]
> @alexunderboots [#99 (comment)](https://github.com/xtrime-ru/antizapret-vpn-docker/issues/99#issuecomment-2424164290)

не помогает, ни на tcp ни на модифицированном-udp

**Comment by gorbunovav** [comment]
@alexunderboots tcp у меня перестал работать какое то время назад (со всех устройств).
Работает только модифицированный UDP.

>не помогает

Остается эта ошибка в контейнере?
>TLS Error: Unroutable control packet received from [AF_INET]ххх.ххх.ххх.ххх:34381 (si=3 op=P_CONTROL_V1)

У меня от нее помогает выключить соединение на Mikrotik на некоторое время.

А с других устройств к этому же серверу подключение работает?

**Comment by alexunderboots** [comment]
> @alexunderboots tcp у меня перестал работать какое то время назад (со всех устройств). Работает только модифицированный UDP.
> 
> > не помогает
> 
> Остается эта ошибка в контейнере?
> 
> > TLS Error: Unroutable control packet received from [AF_INET]ххх.ххх.ххх.ххх:34381 (si=3 op=P_CONTROL_V1)
> 
> У меня от нее помогает выключить соединение на Mikrotik на некоторое время.
> 
> А с других устройств к этому же серверу подключение работает?

Так-же. Но модификация udp не помогает. Ошибка остается. 3 микрота (2 провайдера) ни под каким вариантом подключиться не могут. CHR в цоде по tcp сразу цепляется

**Comment by xtrime-ru** [comment]
Релизнул версию 4. В ней помимо прочего обновил патч для обфускации OPENVPN, теперь уровень обфускации можно менять через ENV переменную без пересборки контейнера. 

Для микротика надо поставить
`OBFUSCATE_TYPE=1` или `OBFUSCATE_TYPE=0` (выключает обфускацию)
